### PR TITLE
B 21213 int

### PR DESCRIPTION
**Pull request by CoryKleinjanCACI**
## Summary

This task updates the PPM closeout queue to show only Needs Closeout ppms. An issue was found where moves with multiple ppms were displaying incorrectly in the queuemoves table, this was also resolved.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Office Application
2. Login as a Service Counselor 
3. Navigate to the Closeout Queue
4. Verify that all moves shown are in the Needs Closeout status

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Comment by antgmann** [comment]
image

Should we remove the option to filter by "Waiting on Customer" if that status will no longer appear in this table?

**Review by MInthavongsay — CHANGES_REQUESTED** [review]
Looks good except for Anthony's comments about removing the waiting on customer status from the drop down. I think that should be removed per ticket description.

File: playwright/tests/office/servicescounseling/servicesCounselingQueueFilters.spec.js [hunk 1]
```diff
@@ -40,12 +40,6 @@ test.describe('Services counselor user', () => {
     });
 
     test('is able to filter moves based on PPM status', async ({ page }) => {
-      // Check for Waiting on customer filter
-      await page.locator('th[data-testid="locator"] > div > input').fill(moveLocator);
-      await page.locator('th[data-testid="locator"] > div > input').blur();
```
[the remark this answers is not in the document]
Comment (CoryKleinjanCACI) [reply playwright/tests/office/servicescounseling/servicesCounselingQueueFilters.spec.js:45]: Whoops. I thought I had removed that.

**Review by antgmann — APPROVED** [review]

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 2]
```diff
@@ -2085,7 +2085,13 @@ func QueueMoves(moves []models.Move, officeUsers []models.OfficeUser, role roles
...
 		for _, shipment := range move.MTOShipments {
 			if shipment.PPMShipment != nil {
-				ppmStatus = shipment.PPMShipment.Status
+				if requestedPpmStatus != nil {
+					if shipment.PPMShipment.Status == *requestedPpmStatus {
+						ppmStatus = shipment.PPMShipment.Status
+					}
+				} else {
+					ppmStatus = shipment.PPMShipment.Status
+				}
```
Comment (ajlusk) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:2094]: Clever way of approaching this. Took me a minute to understand what this was doing. Seems like something that could benefit from a comment and/or backlog number.

**Review by ajlusk — APPROVED** [review]
Looks good! The only status displayed in the closeout queue is needs closeout including for moves that contain other shipments that aren't yet at that stage.

Screenshot 2024-10-09 at 11 39 55 AM

This also displays correctly when viewing as an HQ user.

**Pull request merged**